airflow/operators/sftp_to_gcs.py

```diff
@@ -0,0 +1,219 @@
+    :type sftp_conn_id: str
+    :param delegate_to: The account to impersonate, if any
+    :type delegate_to: str
+    :param mime_type: The mime-type string
+    :type mime_type: str
+    :param gzip: Allows for file to be compressed and uploaded as gzip
+    :type gzip: bool
+    :param move_object: When move object is True, the object is moved instead
+        of copied to the new location. This is the equivalent of a mv command
+        as opposed to a cp command.
+    :type move_object: bool
+
+       :Example:
+
+    The following Operator would copy a single file named
+    ``sales/sales-2017/january.avro`` from the SFTP to the file named
+    ``copied_sales/2017/january-backup.avro`` in the ``data_backup`` bucket ::
+
+        copy_single_file = SFTPToGoogleCloudStorageOperator(
+            task_id="copy_single_file",
+            source_path='sales/sales-2017/january.avro',
+            destination_bucket=`data_backup`,
+            destination_path='copied_sales/2017/january-backup.avro',,
+        )
+
+    The following Operator would copy all the Avro files from ``sales/sales-2017``
+    folder (i.e. with names starting with that prefix) from the SFTP to the
+    ``copied_sales/2017`` folder in the ``data_backup`` bucket. ::
+
+        copy_files = SFTPToGoogleCloudStorageOperator(
+            task_id="copy_files",
+            source_path='sales/sales-2017/*.avro',
+            destination_bucket=`data_backup`,
+            destination_path='copied_sales/2017/',
+        )
+
+    The following Operator would move all the Avro files from ``sales/sales-2017``
+    folder (i.e. with names starting with that prefix) in ``data`` bucket to the
+    same folder in the ``data_backup`` bucket, deleting the original files in the
+    process. ::
+
+        move_files = SFTPToGoogleCloudStorageOperator(
+            task_id="move_files",
+            source_path='sales/sales-2017/*.avro',
+            destination_bucket=`data_backup`,
+            destination_path='copied_sales/2017/',
+            move_object=True,
+        )
+
+    """
+
+    template_fields = ("source_path", "destination_path", "destination_bucket")
+
+    @apply_defaults
+    def __init__(
+        self,
+        source_path: str,
+        destination_bucket: str,
+        destination_path: Optional[str] = None,
+        gcp_conn_id: str = "google_cloud_default",
+        sftp_conn_id: str = "ssh_default",
+        delegate_to: Optional[str] = None,
+        mime_type: str = "application/octet-stream",
+        gzip: bool = False,
+        move_object: bool = False,
+        *args,
+        **kwargs
+    ) -> None:
+        super().__init__(*args, **kwargs)
+
+        self.source_path = source_path
+        self.destination_path = self._set_destination_path(destination_path)
+        self.destination_bucket = self._set_bucket_name(destination_bucket)
+        self.gcp_conn_id = gcp_conn_id
+        self.mime_type = mime_type
```

Comment:
Is this field used somewhere?

Reply:
It should be used in `GoogleCloudStorageHook.upload` however I've missed it. Fixed.

Reply:
It should be used in `GoogleCloudStorageHook.upload`. I will fixed it.